tests/virt/conftest.py

```diff
@@ -5,95 +5,92 @@
-        LOGGER.info("Verify cluster has NUMA")
-        cat_cmd = "cat /etc/kubernetes/kubelet.conf"
-        single_numa_node_cmd = f"{cat_cmd} | grep -i single-numa-node"
-        topology_manager_cmd = f"{cat_cmd} | grep -w TopologyManager"
-        for cmd in (single_numa_node_cmd, topology_manager_cmd):
-            for node in _schedulable_nodes:
-                pod_exec = ExecCommandOnPod(utility_pods=_workers_utility_pods, node=node)
-                out = pod_exec.exec(command=cmd, ignore_rc=True)
-                if not out:
-                    failed_verifications_list.append(f"Cluster does not have {cmd.split()[-1]} enabled")
+        if any(item.get_closest_marker("gpu") for item in request.session.items):
+            LOGGER.info("Verify cluster doesn't have DPDK enabled")
+            if PerformanceProfile(name="dpdk").exists:
+                failed_verifications_list.append("Cluster has DPDK enabled (DPDK is incomatible with NVIDIA GPU)")
 
     def _verify_sriov(_sriov_workers):
-        LOGGER.info("Verify cluster has worker node with SR-IOV card")
-        if not _sriov_workers:
-            failed_verifications_list.append("Cluster doesn't have any SR-IOV workers")
+        if any(item.get_closest_marker("sriov") for item in request.session.items):
+            LOGGER.info("Verify cluster has worker node with SR-IOV card")
+            if not _sriov_workers:
+                failed_verifications_list.append("Cluster does not have any SR-IOV workers")
 
     def _verify_evmcs_support(_schedulable_nodes):
-        LOGGER.info("Verify cluster nodes support VMX cpu fixture")
-        for node in _schedulable_nodes:
-            if not any([
-                label == "cpu-feature.node.kubevirt.io/vmx" and value == "true" for label, value in node.labels.items()
-            ]):
-                failed_verifications_list.append("Cluster doens't have any node that supports VMX cpu feature")
+        if any(item.get_closest_marker("high_resource_vm") for item in request.session.items):
+            LOGGER.info("Verify cluster nodes support VMX cpu fixture")
+            for node in _schedulable_nodes:
+                if not any([
+                    label == "cpu-feature.node.kubevirt.io/vmx" and value == "true"
+                    for label, value in node.labels.items()
+                ]):
+                    failed_verifications_list.append("Cluster does not have any node that supports VMX cpu feature")
 
     def _verify_hugepages_1gi(_workers):
-        LOGGER.info("Verify cluster has 1Gi hugepages enabled")
-        if not any([
-            parse_string_unsafe(worker.instance.status.allocatable["hugepages-1Gi"]) >= parse_string_unsafe("1Gi")
-            for worker in _workers
-        ]):
-            failed_verifications_list.append("Cluster doesn't have hugepages-1Gi")
+        if any(item.get_closest_marker("hugepages") for item in request.session.items):
```

Comment:
You are already checking it at line 28. You might want to move these validation to that section.

Reply:
line 28 is for `high_performace_vm`
i'm not sure it should be combined with hugepages?

Reply:
updated